src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/BaselineController.kt

```diff
@@ -24,22 +27,35 @@ class BaselineController(private val fileManager: FileManager) {
         return SuccessResponse(buildPjnzResponse(fileName, countryName)).toJsonString()
     }
 
-    @GetMapping("/")
+    @PostMapping("/shape/")
     @ResponseBody
-    fun get(): String {
+    fun uploadShape(@RequestParam("file") file: MultipartFile): ResponseEntity<String> {
+        return saveAndValidate(file, FileType.Shape)
+    }
+
+    @GetMapping("/pjnz/")
+    @ResponseBody
+    fun getPJNZ(): String {
 
         // TODO request serialised data for this id from the R API
         // for now just read basic file info from upload dir
         val file = fileManager.getFile(FileType.PJNZ)
 
-        val data = if (file != null) {
+        val pjnzResponse = if (file != null) {
             val fileName = file.name
             val countryName = fileName.split("_").first()
-            mapOf("pjnz" to buildPjnzResponse(fileName, countryName))
+            buildPjnzResponse(fileName, countryName)
         } else {
-            mapOf("pjnz" to null)
+            null
         }
-        return SuccessResponse(data).toJsonString()
+
+        return SuccessResponse(pjnzResponse).toJsonString()
```

Comment:
Originally I was thinking we'd return all baseline data in one endpoint but that doesn't really make sense, so this is now just for pjnz, separate endpoint for shape below